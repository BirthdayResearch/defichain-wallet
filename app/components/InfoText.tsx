import { useThemeContext } from '@contexts/ThemeProvider'
import { MaterialIcons } from '@expo/vector-icons'
import { tailwind } from '@tailwind'
import React from 'react'
import { Text, View } from 'react-native'
import { ThemedProps } from './themed'

interface InfoTextProp extends ThemedProps {
  text: string
}

export function InfoText (props: InfoTextProp): JSX.Element {
  const { isLight } = useThemeContext()
  const {
    light = tailwind('bg-warning-50'),
    dark = tailwind('bg-darkwarning-50')
  } = props

  return (
    <View style={[tailwind('rounded m-4 p-2 flex-row'), (isLight ? light : dark)]}>
      <MaterialIcons name='info' size={14} style={isLight ? tailwind('text-warning-500') : tailwind('text-darkwarning-500')} />
      <Text style={[
        tailwind('text-xs pl-2'),
        isLight ? tailwind('text-gray-600') : tailwind('text-gray-300')
      ]}
      >
        {props.text}
      </Text>
    </View>
  )
}
