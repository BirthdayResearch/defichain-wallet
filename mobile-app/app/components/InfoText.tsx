import { tailwind } from '@tailwind'
import { TextProps } from '.'
import { ThemedIcon, ThemedProps, ThemedText, ThemedView } from './themed'

interface InfoTextProp extends ThemedProps, TextProps {
  text: string
  type?: InfoTextType
}

export type InfoTextType = 'warning' | 'error' | 'success'

export function InfoText (props: InfoTextProp): JSX.Element {
  const {
    type = 'warning',
    style,
    light = tailwind({ 'bg-warning-50 border-warning-200': type === 'warning', 'bg-error-50 border-error-200': type === 'error', 'bg-success-50 border-success-200': type === 'success' }),
    dark = tailwind({ 'bg-darkwarning-50 border-darkwarning-200': type === 'warning', 'bg-darkerror-50 border-darkerror-200': type === 'error', 'bg-success-50 border-success-200': type === 'success' }),
    ...otherProps
  } = props

  return (
    <ThemedView
      style={[
        tailwind('rounded p-2 flex-row border'),
        style
      ]}
      light={light}
      dark={dark}
    >
      <ThemedIcon
        iconType='MaterialIcons'
        name={type === 'success' ? 'check-circle-outline' : type === 'warning' ? 'info' : 'warning'}
        size={14}
        light={tailwind({ 'text-warning-500': type === 'warning', 'text-error-500': type === 'error', 'text-success-500': type === 'success' })}
        dark={tailwind({ 'text-darkwarning-500': type === 'warning', 'text-darkerror-500': type === 'error', 'text-darksuccess-500': type === 'success' })}
      />
      <ThemedText
        style={tailwind('text-xs pl-2 font-medium flex-1')}
        light={tailwind('text-gray-600')}
        dark={tailwind('text-gray-300')}
        {...otherProps}
      >
        {props.text}
      </ThemedText>
    </ThemedView>
  )
}
