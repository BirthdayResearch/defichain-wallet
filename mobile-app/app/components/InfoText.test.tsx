import { render } from '@testing-library/react-native'
import { InfoText } from './InfoText'

jest.mock('@shared-contexts/ThemeProvider')

describe('info text', () => {
  it('should match snapshot', async () => {
    const rendered = render(<InfoText text='foo' />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
