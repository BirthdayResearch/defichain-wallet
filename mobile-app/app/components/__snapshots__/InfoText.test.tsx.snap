// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`info text should match snapshot 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "borderBottomLeftRadius": 4,
          "borderBottomRightRadius": 4,
          "borderBottomWidth": 1,
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
          "borderTopWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
        undefined,
      ],
      Object {
        "backgroundColor": "rgba(255, 250, 244, 1)",
        "borderBottomColor": "rgba(255, 210, 163, 1)",
        "borderLeftColor": "rgba(255, 210, 163, 1)",
        "borderRightColor": "rgba(255, 210, 163, 1)",
        "borderTopColor": "rgba(255, 210, 163, 1)",
      },
    ]
  }
>
  <Text />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Array [
          Object {
            "flexBasis": "0%",
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "MediumFont",
            "fontSize": 12,
            "fontWeight": "500",
            "lineHeight": 16,
            "paddingLeft": 8,
          },
          Object {
            "color": "rgba(82, 82, 82, 1)",
          },
        ],
      ]
    }
    text="foo"
  >
    foo
  </Text>
</View>
`;
